src/components/AddItemForm.js

```diff
@@ -16,25 +17,37 @@ class AddItemForm extends React.Component {
   }
 
   handleChange(event) {
-    this.setState({ name: event.target.value });
+    let name_normalized = this.normalizeUserInput(event.target.value);
+    this.setState({
+      name: event.target.value,
+      name_normalized: name_normalized,
```

Comment:
Not a blocker, you'll often use a short-hand for object keys that map to variables of the same name: 
```js
const normalized_name = this.normalize();
this.setState({
  name: value,
  name_normalized,
});
```

or I might just leave out the variable:
```js
this.setState({
  name: value,
  name_normalized: this.normalize(),
});

```